packages/docs/src/page-configs/ui-elements/data-table/examples/CRUD.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <va-data-table :items="items" :columns="columns" striped>
+    <template #headerAppend>
+      <tr class="table-example--slot">
+        <th colspan="1"><va-input placeholder="name" v-model="addNameModel" /></th>
+        <th colspan="1"><va-input placeholder="username" v-model="addUsernameModel" /></th>
+        <th colspan="1"><va-input placeholder="email" v-model="addEmailModel" /></th>
+        <th colspan="1"><va-button @click="addNewItem()" :disabled="!isNewData">Add new</va-button></th>
+      </tr>
+    </template>
+
+    <template #cell(actions)="{ rowIndex }">
+      <va-button flat icon="edit" @click="openModalToEditItemById(rowIndex)" />
+      <va-button flat icon="delete" @click="deleteItemById(rowIndex)" />
+    </template>
+  </va-data-table>
+
+  <va-modal
+    v-model="showModal"
+    message="Edit item"
+    @ok="editItem()"
+  >
+    <slot>
+      <va-input class="my-3" label="name" v-model="editNameModel" />
+      <va-input class="my-3" label="username" v-model="editUsernameModel" />
+      <va-input class="my-3" label="email" v-model="editEmailModel" />
+    </slot>
+  </va-modal>
+</template>
+
+<script>
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data () {
+    const items = [
+      {
+        name: 'Leanne Graham',
+        username: 'Bret',
+        email: 'Sincere@april.biz',
+      },
+      {
+        name: 'Ervin Howell',
+        username: 'Antonette',
+        email: 'Shanna@melissa.tv',
+      },
+      {
+        name: 'Clementine Bauch',
+        username: 'Samantha',
+        email: 'Nathan@yesenia.net',
+      },
+      {
+        name: 'Patricia Lebsack',
+        username: 'Karianne',
+        email: 'Julianne.OConner@kory.org',
+      },
+    ]
+
+    const columns = [
+      { key: 'name', sortable: true },
+      { key: 'username', sortable: true },
+      { key: 'email', sortable: true },
+      { key: 'actions', width: 130 },
+    ]
+
+    return {
+      items,
+      columns,
+
+      addNameModel: '',
```

Comment:
FIrst off - I would suggest using an object for create and edit.
I.e. 
```
editedUser: null
editedUser: {id: '', name: '', surname: '', email: ''}
```
Same for created user. So resets updates would look differently because of different data structure.

Reply:
Thanks! Fixed it.